docs/reference/contents.md

```diff
@@ -0,0 +1,129 @@
+or [xeus kernels](https://github.com/jupyterlite/xeus/)) relies on the
+[Emscripten filesystem](https://emscripten.org/docs/api_reference/Filesystem-API.html)
+to access their contents. For such case, `@jupyterlite/contents` provides a helper class
+[`DriveFS`](https://jupyterlite.readthedocs.io/en/stable/reference/api/ts/classes/jupyterlite_contents.DriveFS-1.html)
+to be mounted within Emscripten filesystem with a snippet like:
+
+```ts
+  const mountpoint = '/drive';
+  const { FS, PATH, ERRNO_CODES } = /* provided by the empscripten module */;
+  const { baseUrl } = options;
+  const { DriveFS } = await import('@jupyterlite/contents');
+
+  const driveFS = new DriveFS({
+    FS,
+    PATH,
+    ERRNO_CODES,
+    baseUrl, // Website base URL
+    driveName: 'my-drive', // Any name of your choosing
+    mountpoint,
+  });
+  FS.mkdir(mountpoint);
+  FS.mount(driveFS, {}, mountpoint);
+  FS.chdir(mountpoint);
+```
+
+After mounting the drive, the Jupyter Server contents (the one displayed in the
+filebrowser) will be available within the kernel under the folder `/drive`.
+
+The website base URL is required for the `DriveFS` to request its content from the main
+application. Diving into the drive architecture will clarify that.
+
+```{mermaid}
+flowchart LR
+    subgraph main thread
+        direction TB
+        M[Main thread]
+        M --- C[Contents]
+    end
+    subgraph webworker
+        direction TB
+        M -.- K[Kernel]
+        K --- FS[Emscripten FS]
+        FS --- D[DriveFS]
+    end
+    S[Service worker] -.-|BroadcastChannel| M
+    D -.-|REST API| S
+```
+
+Three threads are at play when running a kernel inside JupyterLite:
+
+- The main thread: it executes the main user interface and knows about the filebrowser
+  contents.
+- The kernel web worker: it executes the kernel (e.g. evaluate the code snippet from a
+  notebook sent by the main thread). It mounts a `DriveFS` into the Emscripten
+  filesystem.
+- The service worker: it serves website assets from cache (to work offline). And it also
+  can capture any other network requests.
+
+Assuming the kernel executes the following python snippet code writing into a text file:
+
+```py
+Path("dummy.txt").write_text("Writing on Emscripten filesystem")
+```
+
+Here is a simplification sequence of interaction happening to perform the filesystem
+operation:
+
+```{mermaid}
+sequenceDiagram
+  participant P as Python interpreter
+  participant F as Emscripten FS
+  participant D as DriveFS
+  participant S as Service Worker
+  participant M as Main thread
+  participant C as Contents manager
+
+  P->>+F: Write text into file
+  F->>+D: Call put
+  D->>+S: Send HTTP POST /api/drive
+  S->>+M: Broadcast message via channel
+  M->>+C: Call `save`
+  C-->>-M: None
+  M-->>-S: Response
+  S-->>-D: Response
+  D-->>-F: Return
+  F-->>-P: Done
+```
+
+When the code interact with the filesystem, the emscripten filesystem API is triggered.
```

Comment:
Maybe this can be rephrased to say Emscripten provides a virtual filesystem that simulates a local file system?

And / or link to https://emscripten.org/docs/porting/files/file_systems_overview.html ?